### PR TITLE
Benchmarks

### PR DESCRIPTION
**Pull request by alanlujan91**
## Add Complete Catalogue of Analytically Solvable Benchmark Models

This PR introduces a comprehensive benchmark suite for consumption-savings dynamic programming models with true closed-form analytical solutions.

### 📋 New Files

**`src/skagent/models/benchmarks.py`** - Complete catalogue of 10 analytically solvable discrete-time consumption-savings models:

**Deterministic Models (Perfect Foresight):**
- **D-1**: Two-period log utility with wealth allocation
- **D-2**: Finite horizon log utility with time-varying consumption rates  
- **D-3**: Infinite horizon CRRA with constant marginal propensity to consume
- **D-4**: Blanchard mortality model with survival risk

**Stochastic Models:**
- **U-1**: Hall (1978) random walk consumption with quadratic utility
- **U-2**: CARA utility with Gaussian shocks and certainty equivalence
- **U-3**: Quadratic utility with time-varying interest rates
- **U-4**: Log utility with permanent income shocks (PIH model)
- **U-5**: Epstein-Zin knife-edge case (collapses to CRRA)
- **U-6**: Quadratic utility with habit formation (exact Riccati solution)

**`tests/test_benchmarks.py`** - Research-grade test suite with three validation layers:
- **Layer 1**: Catalogue completeness and model registry structure
- **Layer 2**: Static identity verification of every closed-form formula
- **Layer 3**: Dynamic optimality checks with Euler equations and budget evolution

### 🎯 Key Features

- **True Analytical Solutions**: Each model provides exact closed-form policies, not approximations
- **Comprehensive API**: Functions for model access, calibration retrieval, and policy generation
- **Rigorous Validation**: 32 tests covering formula accuracy, budget constraints, and simulation consistency
- **Research Ready**: Proper mathematical foundations with literature references and theoretical justification

### 🔬 Use Cases

- **Algorithm Benchmarking**: Test numerical methods against known analytical solutions
- **Method Validation**: Verify DP solvers produce correct results on solvable problems  
- **Educational Tool**: Study canonical consumption models with exact solutions
- **Research Foundation**: Build upon proven analytical frameworks

This catalogue represents the complete set of discrete-time consumption-savings DP problems for which the literature has succeeded in deriving true closed-form policies, providing skagent with a gold-standard benchmark suite for numerical method evaluation.

### CODE REVIEW

**Comment by alanlujan91** [comment]
@sbenthall something in maliar.py is not passing? but it passed before...

File: src/skagent/models/benchmarks.py [hunk 1]
```diff
@@ -0,0 +1,1048 @@
...
+#   β/c₂ = λ/R
+
+# Combining: c₂ = βRc₁
+# Budget constraint: c₁ + βRc₁/R = W ⟹ c₁(1+β) = W
+
+# Closed-form solution:
+#   c₁ = W/(1+β)
+#   c₂ = βRW/(1+β)
+
+# This is the foundation for all intertemporal choice theory.
```
Comment (sbenthall) [inline src/skagent/models/benchmarks.py:66]: not sure a comment like this is appropriate -- it's a bit disciplinary specific to economics. there might be intertermporal choice in non-economics fields that are about something else?

File: src/skagent/models/benchmarks.py [hunk 2]
```diff
@@ -0,0 +1,1048 @@
...
+
+    if model_id not in BENCHMARK_MODELS:
+        return {"error": f"Unknown model: {model_id}"}
+
+    try:
+        calibration = get_benchmark_calibration(model_id)
+        analytical_policy = get_analytical_policy(model_id)
+
+        # Generate appropriate test states for each model
+        if model_id == "D-1":
```
Comment (sbenthall) [inline src/skagent/models/benchmarks.py:929]: It's weird to have this general method to have so much that is hard-coded about specific benchmark models. All other information (block, calibration, analytical policy) for each model is handled separately and stored in a dictionary. This makes it easily extensible. This method is _not_ easy to extend, because it involves changing hard-coded stuff in a method.

Is there a way to refactor this so that it refers to a different more general data structure for the test states?

File: src/skagent/models/benchmarks.py [hunk 3]
```diff
@@ -0,0 +1,1048 @@
...
+            "max_consumption": torch.max(
+                analytical_controls[list(analytical_controls.keys())[0]]
+            ).item(),
+            "min_consumption": torch.min(
+                analytical_controls[list(analytical_controls.keys())[0]]
+            ).item(),
+        }
+
+    except Exception as e:
+        return {"error": f"Validation failed: {str(e)}"}
```
Comment (sbenthall) [inline src/skagent/models/benchmarks.py:1018]: Lack of parallel construction in method outputs. If it returns `{'success' : True, ...}` upon success, it should at least return `{'success': False, ...}` on failure, or else this data structure can't be used to evaluate success without further exception handling.

**Comment by sbenthall** [comment]
> @sbenthall something in maliar.py is not passing? but it passed before...

In `test_ann.py` I set the torch seed `torch.manual_seed(10077696)` but I don't think I've done that in `test_maliar.py`.

I'll add that in now. We probably should have a policy that any automated test that uses `torch` should set the seed to ensure consistency.

Otherwise, what's happened is that I'm testing for tolerance on two vectors generated by a neural network that starts with a random seed, which makes it a stochastic test.

**Comment by sbenthall** [comment]
> In `test_ann.py` I set the torch seed `torch.manual_seed(10077696)` but I don't think I've done that in `test_maliar.py`.
> 
> I'll add that in now. We probably should have a policy that any automated test that uses `torch` should set the seed to ensure consistency.

That didn't fix it. My tests are passing locally. Are yours?

It could be related to #56 if the problem is stochastic test failure. But the problem seems to be consistent across all the versions of python and operating systems, which is mysterious (it wouldn't be the random seed in this case).

**Comment by sbenthall** [comment]
Tests pass. Great.

I had a few other comments, but I think the most serious one is about the extensibility of `validate_analytical_solution`
